<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>900</width>
    <height>800</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>RGA Carrier Creator</string>
  </property>
  <widget class="QWidget" name="centralwidget">

   <!-- User Info Section -->
   <widget class="QGroupBox" name="groupBox_user">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>20</y>
      <width>411</width>
      <height>200</height>
     </rect>
    </property>
    <property name="title">
     <string>User Information</string>
    </property>
    <layout class="QFormLayout">
     <item row="0" column="0">
      <widget class="QLabel" name="label_email">
       <property name="text">
        <string>Email:</string>
       </property>
      </widget>
     </item>
     <item row="0" column="1">
      <widget class="QLineEdit" name="email_lineEdit"/>
     </item>
     <item row="1" column="0">
      <widget class="QLabel" name="label_username">
       <property name="text">
        <string>Username:</string>
       </property>
      </widget>
     </item>
     <item row="1" column="1">
      <widget class="QLineEdit" name="username_lineEdit"/>
     </item>
     <item row="2" column="0">
      <widget class="QLabel" name="label_project">
       <property name="text">
        <string>Project:</string>
       </property>
      </widget>
     </item>
     <item row="2" column="1">
      <widget class="QComboBox" name="project_comboBox"/>
     </item>
     <item row="3" column="0">
      <widget class="QLabel" name="label_session">
       <property name="text">
        <string>Session:</string>
       </property>
      </widget>
     </item>
     <item row="3" column="1">
      <widget class="QLineEdit" name="session_lineEdit"/>
     </item>
     <item row="4" column="0">
      <widget class="QLabel" name="label_comments">
       <property name="text">
        <string>Comments:</string>
       </property>
      </widget>
     </item>
     <item row="4" column="1">
      <widget class="QLineEdit" name="comments_lineEdit"/>
     </item>
     <item row="5" column="0">
      <widget class="QLabel" name="label_tags">
       <property name="text">
        <string>Tags:</string>
       </property>
      </widget>
     </item>
     <item row="5" column="1">
      <widget class="QLineEdit" name="tags_lineEdit"/>
     </item>
     <item row="6" column="1">
      <widget class="QPushButton" name="logout_button">
       <property name="text">
        <string>Logout</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>

   <!-- RGA Carrier Info Section -->
   <widget class="QGroupBox" name="groupBox_rga">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>230</y>
      <width>411</width>
      <height>180</height>
     </rect>
    </property>
    <property name="title">
     <string>Create RGA Carrier</string>
    </property>
    <layout class="QFormLayout">
     <item row="0" column="0">
      <widget class="QLabel" name="label_rga_name">
       <property name="text">
        <string>RGA Name:</string>
       </property>
      </widget>
     </item>
     <item row="0" column="1">
      <widget class="QLineEdit" name="lineEdit_rgaName">
       <property name="text">
        <string>RGA000000</string>
       </property>
      </widget>
     </item>
     <item row="1" column="0">
      <widget class="QLabel" name="label_mf_rga_uuid">
       <property name="text">
        <string>MF RGA UUID:</string>
       </property>
      </widget>
     </item>
     <item row="1" column="1">
      <widget class="QLineEdit" name="lineEdit_mf_rga_uuid"/>
     </item>
     <item row="2" column="0">
      <widget class="QLabel" name="label_als_rga_uuid">
       <property name="text">
        <string>ALS RGA UUID:</string>
       </property>
      </widget>
     </item>
     <item row="2" column="1">
      <widget class="QLineEdit" name="lineEdit_als_rga_uuid"/>
     </item>
     <item row="3" column="0">
      <widget class="QPushButton" name="pushButton_new_rga_name">
       <property name="text">
        <string>Get New RGA Name</string>
       </property>
      </widget>
     </item>
     <item row="3" column="1">
      <widget class="QPushButton" name="pushButton_create_rga_crucible">
       <property name="text">
        <string>Add to Crucible</string>
       </property>
      </widget>
     </item>
     <item row="4" column="1">
      <widget class="QPushButton" name="pushButton_create_rga_als">
       <property name="text">
        <string>Add to ALS DB</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>

   <!-- Source Carrier Section -->
   <widget class="QGroupBox" name="groupBox_carrier">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>420</y>
      <width>411</width>
      <height>100</height>
     </rect>
    </property>
    <property name="title">
     <string>Source Carrier</string>
    </property>
    <layout class="QFormLayout">
     <item row="0" column="0">
      <widget class="QLabel" name="label_carrier_uuid_text">
       <property name="text">
        <string>Carrier UUID:</string>
       </property>
      </widget>
     </item>
     <item row="0" column="1">
      <widget class="QLineEdit" name="lineEdit_carrier_uuid"/>
     </item>
     <item row="1" column="0">
      <widget class="QLabel" name="label_carrier_name_text">
       <property name="text">
        <string>Carrier Name:</string>
       </property>
      </widget>
     </item>
     <item row="1" column="1">
      <widget class="QLabel" name="label_carrier_name">
       <property name="text">
        <string>No Carrier Scanned</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>

   <!-- Sample Selection Section -->
   <widget class="QGroupBox" name="groupBox_select">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>530</y>
      <width>411</width>
      <height>150</height>
     </rect>
    </property>
    <property name="title">
     <string>Add Samples to RGA Layout</string>
    </property>
    <layout class="QFormLayout">
     <item row="0" column="0">
      <widget class="QLabel" name="label_select_pos">
       <property name="text">
        <string>RGA Position:</string>
       </property>
      </widget>
     </item>
     <item row="0" column="1">
      <widget class="QComboBox" name="comboBox_select_rgapos"/>
     </item>
     <item row="1" column="0">
      <widget class="QLabel" name="label_select_tf">
       <property name="text">
        <string>Select Sample:</string>
       </property>
      </widget>
     </item>
     <item row="1" column="1">
      <widget class="QComboBox" name="comboBox_select_thinfilm"/>
     </item>
     <item row="2" column="0">
      <widget class="QPushButton" name="pushButton_add_to_rga">
       <property name="text">
        <string>Add to Position</string>
       </property>
      </widget>
     </item>
     <item row="2" column="1">
      <widget class="QPushButton" name="pushButton_add_all">
       <property name="text">
        <string>Add All</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>

   <!-- RGA Parameters Section -->
   <widget class="QGroupBox" name="groupBox_params">
    <property name="geometry">
     <rect>
      <x>450</x>
      <y>20</y>
      <width>411</width>
      <height>100</height>
     </rect>
    </property>
    <property name="title">
     <string>RGA Parameters</string>
    </property>
    <layout class="QFormLayout">
     <item row="0" column="0">
      <widget class="QLabel" name="label_shutter">
       <property name="text">
        <string>Shutter Open (s):</string>
       </property>
      </widget>
     </item>
     <item row="0" column="1">
      <widget class="QSpinBox" name="spinBox_shutter">
       <property name="maximum">
        <number>10000</number>
       </property>
       <property name="value">
        <number>300</number>
       </property>
      </widget>
     </item>
     <item row="1" column="0">
      <widget class="QLabel" name="label_mass_range">
       <property name="text">
        <string>Mass Range (amu):</string>
       </property>
      </widget>
     </item>
     <item row="1" column="1">
      <widget class="QSpinBox" name="spinBox_mass_range">
       <property name="maximum">
        <number>10000</number>
       </property>
       <property name="value">
        <number>300</number>
       </property>
      </widget>
     </item>
    </layout>
   </widget>

   <!-- RGA Layout Grid - Placeholder for 36 positions -->
   <widget class="QGroupBox" name="groupBox_layout">
    <property name="geometry">
     <rect>
      <x>450</x>
      <y>130</y>
      <width>430</width>
      <height>400</height>
     </rect>
    </property>
    <property name="title">
     <string>RGA Carrier Layout (A1-F6)</string>
    </property>
    <widget class="QScrollArea" name="scrollArea">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>30</y>
       <width>410</width>
       <height>360</height>
      </rect>
     </property>
     <property name="widgetResizable">
      <bool>true</bool>
     </property>
     <widget class="QWidget" name="scrollAreaWidgetContents">
      <layout class="QGridLayout" name="gridLayout_positions">
       <!-- Row A -->
       <item row="0" column="0"><widget class="QLabel"><property name="text"><string>A1:</string></property></widget></item>
       <item row="0" column="1"><widget class="QLineEdit" name="lineEdit_tf_A1"/></item>
       <item row="0" column="2"><widget class="QLabel"><property name="text"><string>A2:</string></property></widget></item>
       <item row="0" column="3"><widget class="QLineEdit" name="lineEdit_tf_A2"/></item>
       <item row="0" column="4"><widget class="QLabel"><property name="text"><string>A3:</string></property></widget></item>
       <item row="0" column="5"><widget class="QLineEdit" name="lineEdit_tf_A3"/></item>

       <item row="1" column="0"><widget class="QLabel"><property name="text"><string>A4:</string></property></widget></item>
       <item row="1" column="1"><widget class="QLineEdit" name="lineEdit_tf_A4"/></item>
       <item row="1" column="2"><widget class="QLabel"><property name="text"><string>A5:</string></property></widget></item>
       <item row="1" column="3"><widget class="QLineEdit" name="lineEdit_tf_A5"/></item>
       <item row="1" column="4"><widget class="QLabel"><property name="text"><string>A6:</string></property></widget></item>
       <item row="1" column="5"><widget class="QLineEdit" name="lineEdit_tf_A6"/></item>

       <!-- Row B -->
       <item row="2" column="0"><widget class="QLabel"><property name="text"><string>B1:</string></property></widget></item>
       <item row="2" column="1"><widget class="QLineEdit" name="lineEdit_tf_B1"/></item>
       <item row="2" column="2"><widget class="QLabel"><property name="text"><string>B2:</string></property></widget></item>
       <item row="2" column="3"><widget class="QLineEdit" name="lineEdit_tf_B2"/></item>
       <item row="2" column="4"><widget class="QLabel"><property name="text"><string>B3:</string></property></widget></item>
       <item row="2" column="5"><widget class="QLineEdit" name="lineEdit_tf_B3"/></item>

       <item row="3" column="0"><widget class="QLabel"><property name="text"><string>B4:</string></property></widget></item>
       <item row="3" column="1"><widget class="QLineEdit" name="lineEdit_tf_B4"/></item>
       <item row="3" column="2"><widget class="QLabel"><property name="text"><string>B5:</string></property></widget></item>
       <item row="3" column="3"><widget class="QLineEdit" name="lineEdit_tf_B5"/></item>
       <item row="3" column="4"><widget class="QLabel"><property name="text"><string>B6:</string></property></widget></item>
       <item row="3" column="5"><widget class="QLineEdit" name="lineEdit_tf_B6"/></item>

       <!-- Row C -->
       <item row="4" column="0"><widget class="QLabel"><property name="text"><string>C1:</string></property></widget></item>
       <item row="4" column="1"><widget class="QLineEdit" name="lineEdit_tf_C1"/></item>
       <item row="4" column="2"><widget class="QLabel"><property name="text"><string>C2:</string></property></widget></item>
       <item row="4" column="3"><widget class="QLineEdit" name="lineEdit_tf_C2"/></item>
       <item row="4" column="4"><widget class="QLabel"><property name="text"><string>C3:</string></property></widget></item>
       <item row="4" column="5"><widget class="QLineEdit" name="lineEdit_tf_C3"/></item>

       <item row="5" column="0"><widget class="QLabel"><property name="text"><string>C4:</string></property></widget></item>
       <item row="5" column="1"><widget class="QLineEdit" name="lineEdit_tf_C4"/></item>
       <item row="5" column="2"><widget class="QLabel"><property name="text"><string>C5:</string></property></widget></item>
       <item row="5" column="3"><widget class="QLineEdit" name="lineEdit_tf_C5"/></item>
       <item row="5" column="4"><widget class="QLabel"><property name="text"><string>C6:</string></property></widget></item>
       <item row="5" column="5"><widget class="QLineEdit" name="lineEdit_tf_C6"/></item>

       <!-- Row D -->
       <item row="6" column="0"><widget class="QLabel"><property name="text"><string>D1:</string></property></widget></item>
       <item row="6" column="1"><widget class="QLineEdit" name="lineEdit_tf_D1"/></item>
       <item row="6" column="2"><widget class="QLabel"><property name="text"><string>D2:</string></property></widget></item>
       <item row="6" column="3"><widget class="QLineEdit" name="lineEdit_tf_D2"/></item>
       <item row="6" column="4"><widget class="QLabel"><property name="text"><string>D3:</string></property></widget></item>
       <item row="6" column="5"><widget class="QLineEdit" name="lineEdit_tf_D3"/></item>

       <item row="7" column="0"><widget class="QLabel"><property name="text"><string>D4:</string></property></widget></item>
       <item row="7" column="1"><widget class="QLineEdit" name="lineEdit_tf_D4"/></item>
       <item row="7" column="2"><widget class="QLabel"><property name="text"><string>D5:</string></property></widget></item>
       <item row="7" column="3"><widget class="QLineEdit" name="lineEdit_tf_D5"/></item>
       <item row="7" column="4"><widget class="QLabel"><property name="text"><string>D6:</string></property></widget></item>
       <item row="7" column="5"><widget class="QLineEdit" name="lineEdit_tf_D6"/></item>

       <!-- Row E -->
       <item row="8" column="0"><widget class="QLabel"><property name="text"><string>E1:</string></property></widget></item>
       <item row="8" column="1"><widget class="QLineEdit" name="lineEdit_tf_E1"/></item>
       <item row="8" column="2"><widget class="QLabel"><property name="text"><string>E2:</string></property></widget></item>
       <item row="8" column="3"><widget class="QLineEdit" name="lineEdit_tf_E2"/></item>
       <item row="8" column="4"><widget class="QLabel"><property name="text"><string>E3:</string></property></widget></item>
       <item row="8" column="5"><widget class="QLineEdit" name="lineEdit_tf_E3"/></item>

       <item row="9" column="0"><widget class="QLabel"><property name="text"><string>E4:</string></property></widget></item>
       <item row="9" column="1"><widget class="QLineEdit" name="lineEdit_tf_E4"/></item>
       <item row="9" column="2"><widget class="QLabel"><property name="text"><string>E5:</string></property></widget></item>
       <item row="9" column="3"><widget class="QLineEdit" name="lineEdit_tf_E5"/></item>
       <item row="9" column="4"><widget class="QLabel"><property name="text"><string>E6:</string></property></widget></item>
       <item row="9" column="5"><widget class="QLineEdit" name="lineEdit_tf_E6"/></item>

       <!-- Row F -->
       <item row="10" column="0"><widget class="QLabel"><property name="text"><string>F1:</string></property></widget></item>
       <item row="10" column="1"><widget class="QLineEdit" name="lineEdit_tf_F1"/></item>
       <item row="10" column="2"><widget class="QLabel"><property name="text"><string>F2:</string></property></widget></item>
       <item row="10" column="3"><widget class="QLineEdit" name="lineEdit_tf_F2"/></item>
       <item row="10" column="4"><widget class="QLabel"><property name="text"><string>F3:</string></property></widget></item>
       <item row="10" column="5"><widget class="QLineEdit" name="lineEdit_tf_F3"/></item>

       <item row="11" column="0"><widget class="QLabel"><property name="text"><string>F4:</string></property></widget></item>
       <item row="11" column="1"><widget class="QLineEdit" name="lineEdit_tf_F4"/></item>
       <item row="11" column="2"><widget class="QLabel"><property name="text"><string>F5:</string></property></widget></item>
       <item row="11" column="3"><widget class="QLineEdit" name="lineEdit_tf_F5"/></item>
       <item row="11" column="4"><widget class="QLabel"><property name="text"><string>F6:</string></property></widget></item>
       <item row="11" column="5"><widget class="QLineEdit" name="lineEdit_tf_F6"/></item>
      </layout>
     </widget>
    </widget>
   </widget>

   <!-- Action Buttons -->
   <widget class="QGroupBox" name="groupBox_actions">
    <property name="geometry">
     <rect>
      <x>450</x>
      <y>540</y>
      <width>430</width>
      <height>140</height>
     </rect>
    </property>
    <property name="title">
     <string>Actions</string>
    </property>
    <layout class="QVBoxLayout">
     <item>
      <widget class="QPushButton" name="pushButton_upload_rga">
       <property name="text">
        <string>Upload Samples to Database</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="pushButton_generate_csv">
       <property name="text">
        <string>Generate RGA Input CSV</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="pushButton_clear_rga">
       <property name="text">
        <string>Clear RGA Layout</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>

  </widget>
 </widget>
 <resources/>
 <connections/>
</ui>
